lib/Service/AiIntegrations/AiIntegrationsService.php

```diff
@@ -99,14 +102,64 @@ public function summarizeThread(Account $account, Mailbox $mailbox, $threadId, a
 			$manager->runTask($summaryTask);
 			$summary = $summaryTask->getOutput();
 
-			$this->cache->addSummary($messageIds, $summary);
+			$this->cache->addValue($this->cache->uildUrlKey($messageIds), $summary);
 
 			return $summary;
 		} else {
 			throw new ServiceException('No language model available for summary');
 		}
 	}
 
+	/**
+	 * @return array
+	 */
+	public function SmartReply(Account $account, Mailbox $mailbox, Message $message, string $currentUserId): array {
+		try {
+			$manager = $this->container->get(IManager::class);
+		} catch (\Throwable $e) {
+			throw new ServiceException('Text processing is not available in your current Nextcloud version', 0, $e);
+		}
+		if(in_array(FreePromptTaskType::class, $manager->getAvailableTaskTypes(), true)) {
+			$cachedReplies = $this->cache->getValue('smartReplies_'.$message->getUid());
+			if($cachedReplies) {
+				return explode("|", $cachedReplies);
+			}
+			$client = $this->clientFactory->getClient($account);
+			try {
+				$imapMessage = $this->mailManager->getImapMessage(
+					$client,
+					$account,
+					$mailbox,
+					$message->getUid(), true
+				);
+				if($this->isPersonalEmail($imapMessage)){
+					return [];
```

Comment:
It exits here. Don't we want suggestions for personal emails?

Reply:
You're right, corrected